### PR TITLE
[notification] Channel APIs

### PR DESCRIPTION
**Pull request by qnkhuat**
Epic: https://github.com/metabase/metabase/issues/43822

This PR adds the `channel` table and 5 apis:
- GET /api/channel : channels listing
- GET /api/channel/:id : get a single channel
- PUT /api/channel/:id : update channel
- POST /api/channel : create a channel
- POST /api/channel/test : test channel connection

Tech doc: https://www.notion.so/metabase/Webhook-for-Alert-4f9c7727e06c4c958687c25e1eece21e?pvs=4

### CODE REVIEW

File: enterprise/backend/src/metabase_enterprise/serialization/v2/models.clj [hunk 1]
```diff
@@ -5,7 +5,8 @@
   ["Database"
    "Field"
    "Segment"
-   "Table"])
+   "Table"
+   "Channel"])
```
Comment (piranha) [inline enterprise/backend/src/metabase_enterprise/serialization/v2/models.clj:9]: 😋

File: enterprise/backend/test/metabase_enterprise/advanced_permissions/api/channel_test.clj [hunk 2]
```diff
@@ -0,0 +1,51 @@
+(ns metabase-enterprise.advanced-permissions.api.channel-test
+  (:require
+   [clojure.test :refer :all]
+   [metabase.api.channel-test :as api.channel-test]
+   [metabase.models.permissions :as perms]
+   [metabase.test :as mt]))
+
+(comment
+ ;; to reigster the :metabase-test channel implementation
```
Comment (piranha) [inline enterprise/backend/test/metabase_enterprise/advanced_permissions/api/channel_test.clj:9]: register

File: src/metabase/channel/core.clj [hunk 3]
```diff
@@ -18,10 +18,9 @@
   "Check whether we can connect to a `channel-type` with `detail`.
 
   Returns `true` if can connect to the channel, otherwise return falsy or throw an appropriate exception.
-  In case of failure, to provide a field-specific error message on UI, return or throw an :errors map where key is the
+  o provide a field-specific error message on UI, return or throw an :errors map where key is the
```
Comment (piranha) [inline src/metabase/channel/core.clj:21]: something weird here

**Review by sansolovyov — APPROVED** [review]
I think it's good overall, but ability to list only all channels... This could be a bit too long for a bigger instances maybe? Or are we relying on the front-end to filter them?

**Comment by qnkhuat** [comment]
@piranha yea we can add paging to it but I don't think we have to worry about it now, I don't expect people to have hundreds of channels.

**Pull request merged**